incubator/hnc/internal/reconcilers/object.go

```diff
@@ -382,6 +388,24 @@ func (r *ObjectReconciler) syncPropagated(ctx context.Context, log logr.Logger,
 	return actionNop, nil
 }
 
+func (r *ObjectReconciler) getSelector(ctx context.Context, log logr.Logger, inst *unstructured.Unstructured) labels.Selector {
+	annot := inst.GetAnnotations()
+	selectorstr, present := annot[selectorAnnot]
+	if present == false {
+		return nil
+	}
+	labelSelector, err := v1.ParseToLabelSelector(selectorstr)
+	if err != nil {
+		log.Error(err, "Could not parse selector annotation to labelSelector")
+		return nil
```

Comment:
We need a better way to surface these errors. @yiqigao217 can you please work with Ginny to determine one?

Reply:
Discussed with Yiqi offline and left a `TODO` here. Yiqi will draft a design doc and I'll implement at another PR